File: tests/fixtures/source-windows/typescript/oversized-function.ts
type ReportEntry = {
  title: string;
  archivedAt: string | null;
  createdAt: string;
  owner?: { displayName: string };
};

type ReportServices = {
  loadAccount(id: string): Promise<{ id: string; name: string }>;
  loadEntries(id: string): Promise<readonly ReportEntry[]>;
  recordAccess(id: string, entries: number): Promise<void>;
};

export async function assembleReport(
  accountId: string,
  includeArchived: boolean,
  services: ReportServices,
): Promise<string> {
  const account = await services.loadAccount(accountId);
  const entries = await services.loadEntries(account.id);

  const visible = entries.filter((entry) => {
    return includeArchived || entry.archivedAt === null;
  });

  const ordered = visible.toSorted((left, right) => {
    return left.createdAt.localeCompare(right.createdAt);
  });

  const lines = ordered.map((entry) => {
    const owner = entry.owner?.displayName ?? "unassigned";
    return `${entry.title} — ${owner}`;
  });

  await services.recordAccess(account.id, lines.length);
  return [`Report for ${account.name}`, ...lines].join("\n");
}

File: tests/fixtures/source-windows/go/oversized-function.go
package windows

import (
	"sort"
	"strings"
)

type ReportRow struct {
	Title      string
	Owner      string
	Archived   bool
	CreatedAt  string
}

func BuildReport(rows []ReportRow, includeArchived bool) string {
	visible := make([]ReportRow, 0, len(rows))
	for _, row := range rows {
		if includeArchived || !row.Archived {
			visible = append(visible, row)
		}
	}

	sort.Slice(visible, func(left, right int) bool {
		return visible[left].CreatedAt < visible[right].CreatedAt
	})

	lines := make([]string, 0, len(visible))
	for _, row := range visible {
		owner := row.Owner
		if owner == "" {
			owner = "unassigned"
		}
		lines = append(lines, row.Title+" — "+owner)
	}

	return strings.Join(lines, "\n")
}
